modules/apps/headless/headless-admin-user/headless-admin-user-impl/src/main/java/com/liferay/headless/admin/user/internal/resource/v1_0/UserAccountResourceImpl.java

```diff
@@ -546,6 +546,9 @@ public UserAccount patchUserAccount(
 
 		ServiceContext serviceContext = _createServiceContext(userAccount);
 
+		_userService.updateExternalReferenceCode(
```

Comment:
Please move this below the .updateUser method and reassign the result to the user variable. I think also this is not correct if I don't pass the ERC in the body because it will reset the field; we should instead use something like 
```
GetterUtil.getString(userAccount.getExternalReferenceCode(),user.getExternalReferenceCode())
```
What do you think?

Reply:
I do agree using the GetterUtil approach, but I'm not able to change the order, when I tested I received a exception like this:

2024-01-23 14:25:17.475 ERROR [http-nio-8080-exec-4][ExceptionMapper:59] null
org.hibernate.StaleObjectStateException: Row was updated or deleted by another transaction (or unsaved-value mapping was incorrect) : [com.liferay.portal.model.impl.UserImpl#32815]
	at org.hibernate.event.internal.DefaultMergeEventListener.entityIsDetached(DefaultMergeEventListener.java:341) ~[hibernate-core.jar:5.6.7.Final]
	at org.hibernate.event.internal.DefaultMergeEventListener.onMerge(DefaultMergeEventListener.java:172) ~[hibernate-core.jar:5.6.7.Final]
	at org.hibernate.event.internal.DefaultMergeEventListener.onMerge(DefaultMergeEventListener.java:70) ~[hibernate-core.jar:5.6.7.Final]
	at org.hibernate.event.service.internal.EventListenerGroupImpl.fireEventOnEachListener(EventListenerGroupImpl.java:107) ~[hibernate-core.jar:5.6.7.Final]
	at org.hibernate.internal.SessionImpl.fireMerge(SessionImpl.java:829) ~[hibernate-core.jar:5.6.7.Final]
	at org.hibernate.internal.SessionImpl.merge(SessionImpl.java:816) ~[hibernate-core.jar:5.6.7.Final]
	at com.liferay.portal.dao.orm.hibernate.SessionImpl.merge(SessionImpl.java:329) ~[portal-impl.jar:?]
	at com.liferay.portal.service.persistence.impl.UserPersistenceImpl.updateImpl(UserPersistenceImpl.java:8998) ~[portal-impl.jar:?]
	at com.liferay.portal.service.persistence.impl.UserPersistenceImpl.updateImpl(UserPersistenceImpl.java:88) ~[portal-impl.jar:?]
	at com.liferay.portal.kernel.service.persistence.impl.BasePersistenceImpl.update(BasePersistenceImpl.java:785) ~[portal-kernel.jar:?]
	at com.liferay.portal.service.base.UserLocalServiceBaseImpl.updateUser(UserLocalServiceBaseImpl.java:482) ~[portal-impl.jar:?]
	at com.liferay.portal.service.impl.UserLocalServiceImpl.updateExternalReferenceCode(UserLocalServiceImpl.java:4497) ~[portal-impl.jar:?]
	at com.liferay.portal.service.impl.UserLocalServiceImpl.updateExternalReferenceCode(UserLocalServiceImpl.java:4479) ~[portal-impl.jar:?]

Reply:
Ok, I don't like it but at the moment keep it before